src/codegen/encoding/mod.rs

```diff
@@ -51,13 +53,16 @@ pub(super) trait AbiEncoding {
 
     /// Some types have sizes that are specific to each encoding scheme, so there is no way to generalize.
     fn get_encoding_size(&self, expr: &Expression, ty: &Type, ns: &Namespace) -> Expression;
+
+    /// Returns if the we are packed encoding
+    fn is_packed(&self) -> bool;
 }
 
 /// This function should return the correct encoder, given the target
-pub(super) fn create_encoder(ns: &Namespace) -> impl AbiEncoding {
+pub(super) fn create_encoder(ns: &Namespace, packed: bool) -> Box<dyn AbiEncoding> {
     match &ns.target {
-        Target::Solana => BorshEncoding::new(),
-        _ => unreachable!("Other types of encoding have not been implemented yet"),
+        Target::Solana => Box::new(BorshEncoding::new(packed)),
+        _ => Box::new(ScaleEncoding::new(packed)),
```

Comment:
```suggestion
        Target::Solana => Box::new(BorshEncoding::new(packed)),
        Target::Substrate { .. } => Box::new(ScaleEncoding::new(packed)),
        _ => unimplemented!("ABI encoder is not implemented yet"),
```

Using the scale encoder for everything that is not Solana is technically not correct? It will also make it easier to catch up when implementing a new backend.

Reply:
I see what you mean, but if I do that, tests will fail because, although we do not support EVM yet, we have code generation tests for a fake EVM target, and it needs the `Instr::AbiEncode` in the CFG.

Reply:
Any suggestions?
@seanyoung and @xermicus ?

Reply:
Right now we don't have an eth abi encoder, doesn't matter which one you pick. However it does need a big comment.